### ISSUE TITLE
MedievalOak barracks and its towers are built up and upgraded in a strange way

### ISSUE THREAD

**Issue by Krisztian-Hosszu** [ldtteam/minecolonies#5139]
I did try to check if this might already be fixed but couldn't verify for sure, so sorry if its a duplicate.

Version: 
structurize-0.10.315-ALPHA
minecolonies-1.15.2-0.11.948-universal

### Expected behavior
I have built a medievaloak type barracks in my world and started upgrading it.
I would expect the barracks towers to be built and upgraded in a way that it fits nicely with the rest of the barracks building (at least when you upgrade the towers to the same level as the barracks)

### Actual behaviour
I have experienced a number of strange blocks after building/upgrading the towers that make me think they weren't intended to be built that way:
**After upgrading the first tower to lvl2 in a lvl2 barracks:**
![barracks_tower1_lvl2](https://user-images.githubusercontent.com/10154329/83545130-12f6c780-a4ff-11ea-85fd-3030b98ea200.png)
A barrel has been placed in the walkway on the ground floor of the first tower after upgrading the tower to lvl 2. On the top you can see cobblestone stairs, this tower has cobblestone stairs on 3 sides and oak stars on the opposite side when viewed from this angle (outside), I'm mentioning this only because I had the feeling the issue might be incorrect rotation placement of the tower(s) and think the wooden stairs part should be the one facing this direction.
![barracks_tower1_lvl2_bedroom](https://user-images.githubusercontent.com/10154329/83545775-f9a24b00-a4ff-11ea-9283-7be3963f0b97.png)
The basement/bedroom part, I think the excess dirt hasn't been cleared from the room after finishing the building
![barracks_tower1_lvl2_ladder](https://user-images.githubusercontent.com/10154329/83545915-32dabb00-a500-11ea-852e-5230f9bd8902.png)
The passage with the ladder leading to the downstairs bedroom was blocked by a dirt/grass block
![barracks_tower1_lvl2_passage](https://user-images.githubusercontent.com/10154329/83546029-5c93e200-a500-11ea-98cc-696c224d02a9.png)
The passage to main room with the barracks building block in it was blocked by a wooden slab
**After building the second tower in a lvl2 barracks:**
![barracks_tower2_lvl1](https://user-images.githubusercontent.com/10154329/83546477-ffe4f700-a500-11ea-9a1a-e910e44b015b.png)
There have been some floating wooden slabs on the wall upstairs. The rest of similar slabs were all placed on top of Oak pillars throughout the build
**After upgrading the second tower to lvl2 in a lvl2 barracks:**
![barracks_tower2_lvl2_passage](https://user-images.githubusercontent.com/10154329/83546647-49354680-a501-11ea-9ba6-3600d38f6626.png)
The passage to main room with the barracks building block in it was blocked by a wooden slab (same as the first tower blocked it from the other side) These 2 upgrades sealed off the main room making it unreachable.
It seems like the tower wall is blocking passage where it previously lead through the tower to the main room and on the outer side there is no wall, but what seems like should connect to the passage coming from the left.
![barracks_tower2_lvl2_bedroom](https://user-images.githubusercontent.com/10154329/83547410-73d3cf00-a502-11ea-8dfc-d4963ea7d2a8.png)
In the second tower, a bedroom downstairs was also built but there is no passage leading down (no ladder was built) and excess dirt hasn't been removed from there and also it isn't lit up in any way (You can see the other towers bedroom on the image as well)
**For reference:**
![barracks facing](https://user-images.githubusercontent.com/10154329/83547641-d0cf8500-a502-11ea-9a6d-9e99df250af6.png)
This image is just showing the barracks building block, I'm facing East, so I'm assuming the block is facing west.
![barracks_tower1_facing](https://user-images.githubusercontent.com/10154329/83547951-40de0b00-a503-11ea-81c6-c018adf36b37.png)
Tower 1, I'm facing North so assuming the block is facing South
![barracks_tower2_facing](https://user-images.githubusercontent.com/10154329/83548012-61a66080-a503-11ea-8544-28226102709a.png)
Tower 2, I'm facing South so assuming the block is facing North

I have not manually touched/rotated/destroyed/replaced the tower blocks that have been put down by the builder when building/upgrading the barracks.
Furthermore I'd like to mention as it might be related, that when I previously tried to reposition a lvl3 building through the Build options (after realizing too late that I placed it 1 block closer than I wanted), I've experienced that however I rotated the building after trying to reposition it always ended up facing South, so had to completely remove and rebuild the building from lvl 1 to 3 again to fix that. (This might be another issue though)

### Steps to reproduce the problem
1. Build a medievaloak type barracks with the building block facing west as shown on the above images.
2. Build the first tower that is placed by the builder
3. Upgrade the barracks
4. Upgrade the first tower to lvl 2
5. Build the second tower that is placed by the builder
6. Upgrade the second tower to lvl 2

### Logs: Latest.log and crashlog at https://gist.github.com/
There has been no crash, no logs available.
- latest.log:
- crashlog:

**Comment by Krisztian-Hosszu** [comment]
Okay, I've just seen the Schematic issues pinned item and there are some comments under there about other types of barracks with the same issue: https://github.com/ldtteam/minecolonies/issues/5006

**Comment by Ampaex** [comment]
I can confirm it also happens in medieval spruce.

**Comment by Raycoms** [comment]
should be fixed in newest

**Comment by Krisztian-Hosszu** [comment]
Do I need to rebuild the whole barracks to fix it after updating to newest or is there a faster way? Does the repair building option fix the towers facing this way?

**Comment by Raycoms** [comment]
you might have to remove the tower, then repair the barracks.
